python/hsfs/feature_group.py

```diff
@@ -1675,6 +1698,151 @@ def insert(
             ge_report.to_ge_type() if ge_report is not None else None,
         )
 
+    def multi_part_insert(
+        self,
+        features: Union[
+            pd.DataFrame,
+            TypeVar("pyspark.sql.DataFrame"),  # noqa: F821
+            TypeVar("pyspark.RDD"),  # noqa: F821
+            np.ndarray,
+            List[list],
+        ] = None,
+        overwrite: Optional[bool] = False,
+        operation: Optional[str] = "upsert",
+        storage: Optional[str] = None,
+        write_options: Optional[Dict[str, Any]] = {},
+        validation_options: Optional[Dict[str, Any]] = {},
+    ) -> Tuple[Optional[Job], Optional[ValidationReport]]:
+        """Get FeatureGroupWriter for optimized multi part inserts or call this method
+        to start manual multi part optimized inserts.
+
+        In use cases where very small batches (1 to 1000) rows per Dataframe need
+        to be written to the feature store repeatedly, it might be inefficient to use
+        the standard `feature_group.insert()` method as it performs some background
+        actions to update the metadata of the feature group object first.
+
+        For these cases, the feature group provides the `multi_part_insert` API,
+        which is optimized for writing many small Dataframes after another.
+
+        There are two ways to use this API:
+        !!! example "Python Context Manager"
+            Using the Python `with` syntax you can acquire a FeatureGroupWriter
+            object that implements the same `multi_part_insert` API.
+            ```python
+            feature_group = fs.get_or_create_feature_group("fg_name", version=1)
```

Comment:
Is there an assumption that users call `save()` before using the `multi_part_insert()` method?